lib/Controller/RadioApiController.php

```diff
@@ -201,4 +203,26 @@ public function resetAll() {
 		$this->businessLayer->deleteAll($this->userId);
 		return new JSONResponse(['success' => true]);
 	}
+
+	/**
+	* get radio metadata from stream
+	*
+	* @NoAdminRequired
+	* @NoCSRFRequired
+	*/
+
+	public function getRadioStreamData(int $id) {
+		try {
+			$response = "";
+			$station = $this->businessLayer->find($id, $this->userId);
+			$stapi = $station->toAPI();
```

Comment:
One more minor thing. Converting the station to the "API format" here is quite pointless. We are interested only about the stream URL and that may be obtained with `$station->getStreamUrl()`. It should be safe to assume this to be non-null since one cannot create a station without passing a stream URL.